Modified the isisVarInit.py script

rbeyer opened the pull request:
## Description
- Removed the '3' from its filename
- Altered so that it writes out `isis-activate.*` and `isis-deactivate.*` files instead of the generic `env_vars.*` filenames.
- Removed the the tremendous amount of leading whitespace written to the files.
- Add a report of the version information on conda activation.  Purely cosmetic.
- General code cleanup.
- Also changes the name of the script in the README instructions.

## Related Issue
- Now uses Pathlib and eliminates the weird assert, so might close #3847 
- Writes out better names, and would close #3848 

## Motivation and Context
I couldn't stand all the leading spaces, and since I was in the code already, I also addressed the above issues.

## How Has This Been Tested?
I copied this script into the $CONDA_PREFIX/scripts/ directory of a ISIS 4.1.0 environment, and it seemed to do all the right things in my zsh shell, but did not test csh or fish.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/). **Can't check, this URL is still 404ing!**
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

rbeyer committed:
Removed version number from filename.

rbeyer committed:
feat: Changed the name of the VarInit script, changed how it worked
internally, and made it write out isis-activate.* and isis_deactivate.*
files.

jessemapel commented on isis/scripts/isisVarInit.py:
```diff
@@ -0,0 +1,146 @@
+csh = dict(
+    extension=".csh",
+    shebang="#!/usr/bin/env csh",
+    activate="setenv {}={}",
```
setenv doesn't use an =, this should be `setenv {} {}`

rbeyer replied:
Thanks, I did mention that I didn't test it.

jessemapel commented on isis/scripts/isisVarInit.py:
```diff
@@ -0,0 +1,146 @@
+
+
+def mkdir(p: Path):
+    # This just wraps and reports on the directory creation:
```
Convert this to a proper doc-string to match the other helper functions.

rbeyer replied:
I did.  I also altered this to return a string instead of None.  I don't like functions calling print() (unless I'm debugging).

rbeyer committed:
Fix: Now has correct csh syntax, updated mkdir() with a proper docstring
and to return a string (only main should print(), functions shouldn't or
should log, but that's overkill for this).
